### ISSUE TITLE
Garbled output at PowerShell when I use redirect.

### ISSUE THREAD

**Issue by frg2089** [dotnet/sdk#32013]
### Describe the bug

When I use PowerShell in a Chinese environment to redirect the output of a dotnet program, I get some garbled.

It can also happen in the pipeline.

I tested MSBuild for Visual Studio, MSBuild for .Net Framework and other applications. Only dotnet does this.

And I tested PowerShell 7.3, Windows PowerShell 5.1 and Command Prompt. this only happens on PowerShell 7.3 and Windows PowerShell 5.1.

### To Reproduce

1. Set your Windows Display Language is Chinese.
2. ```powershell
   dotnet -? > stdout.log # Redirect standard output into stdout.log file.
   # or
   dotnet -? | clip # Copy the standard output into your clipboard.
   ```

### Further technical details

```
.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.203\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  6.0.406 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

```
Windows Display Language  
![Windows Display Language](https://user-images.githubusercontent.com/42184238/234178182-a646d809-b6ab-4a91-b5c3-7f8a235b3b01.png)

Direct run in PowerShell 7.3  
![Direct run in PowerShell 7.3](https://user-images.githubusercontent.com/42184238/234178234-d24aafaa-7733-433b-90f9-54b308dfc087.png)

Redirect into file  
![Redirect into file](https://user-images.githubusercontent.com/42184238/234178341-89545594-f2fd-4229-a667-f5aef06f5e8c.png)

**Comment by frg2089** [comment]
It also in .Net 8.0

![image](https://user-images.githubusercontent.com/42184238/234180657-64bfcfbc-53f7-4c49-9374-e053b2ebe195.png)

**Comment by frg2089** [comment]
This garble looks like it is caused by UTF-8 encoded content being saved incorrectly as GB18030 (GBK or GB2312).

**Comment by frg2089** [comment]
Maybe it's a problem here.

https://github.com/dotnet/sdk/blob/abc94993c152aa557f953b897abaaf38dff61f86/src/Cli/dotnet/Program.cs#L33-L41

It looks like the code here causes being redirected when the current OS is not iOS, Android, TV OS and browser to force UTF-8 encoding, while PowerShell running on Windows will treat them as if they were ANSI.

**Comment by s1iver1ining** [comment]
yes, it IS because the encoding issues, but i don't know how to solve it.
My problem is using redirection in cmd prompt when I was learning <c primer plus>chapter8.
But both my input and output don't have ANY Chinese character, but it still went wrong. Two lines were replaced by garbled codes! For No reason!! This is just like a black box, and I have nothing to do about it, so powerless...
![3-14](https://github.com/dotnet/sdk/assets/156559521/b6285eb9-1b44-4d85-a42c-606d1ef1be56)
